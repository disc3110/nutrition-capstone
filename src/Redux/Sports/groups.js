/* eslint-disable no-param-reassign */
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

export const fetchGroups = createAsyncThunk('sports/fetchGroups', async () => {
  const response = await axios.get('https://sports.api.decathlon.com/groups');
  return response.data.data;
});

export const sportsGroupsSlice = createSlice({
  name: 'groups',
  initialState: { entities: [], loading: 'idle' },
  reducers: {
  },
  extraReducers: (builder) => {
    builder
      .addCase(fetchGroups.pending, (state) => {
        state.loading = 'pending';
      })
      .addCase(fetchGroups.fulfilled, (state, action) => {
        state.entities = [...action.payload];
        state.loading = 'idle';
      });
  },
});

export default sportsGroupsSlice.reducer;
