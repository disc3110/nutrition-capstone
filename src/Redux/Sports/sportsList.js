/* eslint-disable no-param-reassign */
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

export const fetchSports = createAsyncThunk('sports/fetchSports', async () => {
  const response = await axios.get('https://sports.api.decathlon.com/sports');
  return response.data.data;
});

export const sportsListSlice = createSlice({
  name: 'sports',
  initialState: { entities: [], loading: 'idle' },
  reducers: {
  },
  extraReducers: (builder) => {
    builder
      .addCase(fetchSports.pending, (state) => {
        state.loading = 'pending';
      })
      .addCase(fetchSports.fulfilled, (state, action) => {
        state.entities = [...action.payload];
        state.loading = 'idle';
      });
  },
});

export default sportsListSlice.reducer;
