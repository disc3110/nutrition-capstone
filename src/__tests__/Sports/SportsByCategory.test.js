import React from 'react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Sports from '../../components/SportsByCategory';
import store from '../../redux/configureStore';

describe('Check if all components are rendered', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Provider store={store}><Sports /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
