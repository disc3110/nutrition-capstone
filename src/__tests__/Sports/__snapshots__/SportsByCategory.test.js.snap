// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if all components are rendered renders correctly 1`] = `
<div
  className="d-flex list-item"
>
  <p
    className="mx-4"
  />
  <img
    alt=""
    className="icon text-end"
  />
</div>
`;
